aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -384,6 +374,153 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return result;
 }
 
+Tensor clamp_with_tensors(const Tensor& self, const Tensor& min, const Tensor& max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_with_tensors_out(result, self, min, max);
+}
+
+Tensor clamp_with_tensors_max(const Tensor& self, const Tensor& max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_with_tensors_max_out(result, self, max);
+}
+
+Tensor clamp_with_tensors_min(const Tensor& self, const Tensor& min) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_with_tensors_min_out(result, self, min);
+}
+
+Tensor& clamp_with_tensors_(Tensor& self, const Tensor& min, const Tensor& max) {
+  return at::clamp_with_tensors_out(self, self, min, max);
+}
+
+Tensor& clamp_with_tensors_out(
+    Tensor& result,
+    const Tensor& self,
+    const Tensor& min,
+    const Tensor& max) {
+  TORCH_CHECK(!self.is_complex(), "clamp is not yet implemented for complex tensors.");
+  if (min.defined() && max.defined()) {
+    TORCH_CHECK(self.device().type() == DeviceType::CPU || self.device().type() == DeviceType::CUDA,
+                "clamp only supports CPU and CUDA device type, got: ", self.device().type());
+    TORCH_CHECK(self.layout() == Layout::Strided,
+                "clamp only supports strided layout, got: ", self.layout());
+
+    // We clamp max tensor to make sure that it is not smaller than min
+    // This is necessary to fix the divergent behaviour in AVX and base CPU kernels
+    // https://github.com/pytorch/pytorch/pull/32587#issuecomment-591389582
+    auto adjusted_max = at::clamp_with_tensors_min(max, min);
```

Comment:
I realize I'm the one who suggested this as a possible fix, but reviewing more closely now I'm worried about it because it causes divergent behavior with NumPy and may impact performance.

If we look at NumPy's implementation (https://github.com/numpy/numpy/blob/e94cec800304a6a467cf90ce4e7d3e207770b4b4/numpy/core/src/umath/clip.c.src#L78) it appears to be the same as our AVX code, and I think we should make our other implementation consistent with it, too.

Reply:
Resolved this by making all backends implement the `min(max_vec, max(min_vec, x)` formula.